### PR TITLE
fix for isRegistered always true

### PR DESCRIPTION
**Pull request by Maar-io**
* When isRegistered() is called with enabled external database (Shiden only) it will always result with true. The problem is that 0x000...000 will have length>0

* Problem fixed with this PR is check for register() when ss58 address is 0.

* set isRegistered() to be view function. Remore re-register from the isRegistered() call

- [x]  unit test updated

### CODE REVIEW

**Comment by niklabh** [comment]
isRegistered has become a transaction and requiring opening metamask. It should remain a view function and should not change

File: contract/shiden/contracts/AstarBaseV4.sol [hunk 1]
```diff
@@ -169,6 +171,12 @@ contract AstarBaseV4
             // register to avoid check in external astarbase next time
             registerExecute(evmAddress, ss58PublicKey);
         }
+        else{
+            // if ss58PublicKey32 was 0, it will be encoded as 0x000...000
+            // this will cause chack in isRegister to show len(0)>0 
+            delete ss58PublicKey;
```
Comment (niklabh) [inline contract/shiden/contracts/AstarBaseV4.sol:177]: We cannot modify blockchain state in isRegistered function. It should remain a view function

**Review by niklabh — CHANGES_REQUESTED** [review]
make isRegistered a view method

File: contract/shiden/contracts/AstarBaseV4.sol (hunk 1 above)
Comment (Maar-io) [reply contract/shiden/contracts/AstarBaseV4.sol:177]: you are right. But the reason that this method needs signing is in call to register()
this delete is for the local variable

Comment (niklabh) [reply contract/shiden/contracts/AstarBaseV4.sol]: Need a view function to show user is registered or not.

Comment (Maar-io) [reply contract/shiden/contracts/AstarBaseV4.sol]: done

**Comment by niklabh** [comment]
working fine

**Review by niklabh — APPROVED** [review]

**Pull request merged**